must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -94,6 +96,9 @@ protected void addComputedTypeAnnotations(Tree tree, AnnotatedTypeMirror type, b
     if (TypesUtils.isPrimitiveOrBoxed(type.getUnderlyingType())) {
       type.replaceAnnotation(BOTTOM);
     }
+    if (isDeclaredInTryWithResources(TreeUtils.elementFromTree(tree))) {
+      type.replaceAnnotation(BOTTOM);
```

Comment:
Is it possible to define a class that implements `Closeable` and also has additional `@MustCall` requirements?  Admittedly a corner case, but if it's possible, we may want to tweak this logic to not set the type to bottom, but instead to just remove the `close` method from the type.  Thoughts?

Reply:
Ah - you're right. I'll fix this.

Reply:
@msridhar I've rewritten to handle this case, and I added a stub for one of the failures in the tests. If we remove the other test, we can merge this if you all are okay with it.